common/fee/models.go

```diff
@@ -0,0 +1,183 @@
+)
+
+func IsBumpErr(err error) bool {
+	return err != nil && (errors.Is(err, ErrBumpFeeExceedsLimit) || errors.Is(err, ErrBump) || errors.Is(err, ErrConnectivity))
+}
+
+// CalculateBumpedFee will increase the price and apply multiplier to the fee limit
+func CalculateBumpedFee(cfg feetypes.BumpConfig, lggr logger.SugaredLogger, currentfeePrice, originalfeePrice *big.Int, originalFeeLimit uint32, maxFeePrice *big.Int) (*big.Int, uint32, error) {
+	feePrice, err := bumpFeePrice(cfg, lggr, currentfeePrice, originalfeePrice, maxFeePrice)
+	if err != nil {
+		return nil, 0, err
+	}
+	chainSpecificFeeLimit := ApplyMultiplier(originalFeeLimit, cfg.LimitMultiplier())
+	return feePrice, chainSpecificFeeLimit, nil
+}
+
+// bumpfeePrice computes the next fee price to attempt as the largest of:
+// - A configured percentage bump (FeeEstimator.BumpPercent) on top of the baseline price.
+// - A configured fixed amount of Unit (FEE_PRICE_Unit) on top of the baseline price.
+// The baseline price is the maximum of the previous fee price attempt and the node's current fee price.
+func bumpFeePrice(cfg feetypes.BumpConfig, lggr logger.SugaredLogger, currentfeePrice, originalfeePrice, maxFeePriceInput *big.Int) (*big.Int, error) {
+	maxFeePrice := getMaxFeePrice(maxFeePriceInput, cfg.PriceMax()) // Make a wrapper config
+	bumpedFeePrice := bumpFeePriceByPercentage(originalfeePrice, cfg.BumpPercent(), cfg.BumpMin())
+
+	// Update bumpedFeePrice if currentfeePrice is higher than bumpedFeePrice and within maxFeePrice
+	bumpedFeePrice = maxBumpedFee(lggr, currentfeePrice, bumpedFeePrice, maxFeePrice, "fee price")
+
+	if bumpedFeePrice.Cmp(maxFeePrice) > 0 {
+		return maxFeePrice, errors.Wrapf(ErrBumpFeeExceedsLimit, "bumped fee price of %s would exceed configured max fee price of %s (original price was %s). %s",
+			bumpedFeePrice.String(), maxFeePrice, originalfeePrice.String(), label.NodeConnectivityProblemWarning)
+	} else if bumpedFeePrice.Cmp(originalfeePrice) == 0 {
+		// NOTE: This really shouldn't happen since we enforce minimums for
+		// FeeEstimator.BumpPercent and FeeEstimator.BumpMin in the config validation,
+		// but it's here anyway for a "belts and braces" approach
+		return bumpedFeePrice, errors.Wrapf(ErrBump, "bumped fee price of %s fee unit is equal to original fee price of %s fee unit."+
+			" ACTION REQUIRED: This is a configuration error, you must increase either "+
+			"FeeEstimator.BumpPercent or FeeEstimator.BumpMin", bumpedFeePrice.String(), originalfeePrice.String())
+	}
+	return bumpedFeePrice, nil
+}
+
+func getMaxFeePrice(userSpecifiedMax, maxFeePrice *big.Int) *big.Int {
+	return FeePriceLimit(userSpecifiedMax, maxFeePrice)
+}
+
+func bumpFeePriceByPercentage(originalFeePrice *big.Int, feeBumpPercent uint16, feeBumpUnits *big.Int) *big.Int {
+	return max(new(big.Int).Add(originalFeePrice, feeBumpUnits), addPercentage(originalFeePrice, feeBumpPercent))
```

Comment:
Comment? Or maybe just reformat so it reads more plainly?
```suggestion
func bumpFeePriceByPercentage(originalFeePrice *big.Int, feeBumpPercent uint16, feeBumpUnits *big.Int) *big.Int {
	return max(
		new(big.Int).Add(originalFeePrice, feeBumpUnits), 
		addPercentage(originalFeePrice, feeBumpPercent),
	)
```

Reply:
thanks for the reformat. It looks much cleaner